data-serving/data-service/src/controllers/preprocessor.ts

```diff
@@ -102,58 +101,82 @@ export const setBatchUpsertRevisionMetadata = async (
             ]),
     );
 
-    // Get sourceId:sourceEntryId for cases that are included in the batch
-    // request but that won't be modified (because the request case is identical
-    // to the existing one).
-    const unchangedCaseIdSet = request.body.cases.reduce(
-        // TODO: Type request Cases.
-        // eslint-disable-next-line @typescript-eslint/no-explicit-any
-        (unchangedIds: Set<string>, c: any) => {
-            if (c.caseReference?.sourceId && c.caseReference?.sourceEntryId) {
-                const existingCase = existingCasesByCaseRefCombo.get(
-                    c.caseReference.sourceId +
-                        ':' +
-                        c.caseReference.sourceEntryId,
-                );
-                if (existingCase !== undefined && existingCase.equalsJSON(c)) {
-                    unchangedIds.add(existingCase._id.toString());
-                }
+    const unchangedCaseIdSet = new Set();
+    request.body.cases = request.body.cases.filter((c: any) => {
+        if (c.caseReference?.sourceId && c.caseReference?.sourceEntryId) {
+            const existingCase = existingCasesByCaseRefCombo.get(
+                c.caseReference.sourceId + ':' + c.caseReference.sourceEntryId,
+            );
+            if (existingCase !== undefined && existingCase.equalsJSON(c)) {
+                unchangedCaseIdSet.add(existingCase._id.toString());
+                return false;
             }
-            return unchangedIds;
-        },
-        new Set(),
-    );
+        }
+        return true;
+    });
 
     // Store the unchanged IDs for future middleware.
     response.locals.unchangedCaseIdSet = unchangedCaseIdSet;
+
+    next();
+};
+
+// Set appropriate values for the revision metadata and uploadids fields.
+export const setBatchUpsertFields = async (
+    request: Request,
+    response: Response,
+    next: NextFunction,
+): Promise<void> => {
+    const curatorEmail = request.body.curator.email;
+
+    // Find and map existing cases by sourceId:sourceEntryId.
+    const existingCasesByCaseRefCombo = new Map(
+        (await findCasesWithCaseReferenceData(request))
+            .filter((c) => c && c.caseReference)
+            .map((c) => [
+                c.caseReference.sourceId + ':' + c.caseReference.sourceEntryId,
+                c,
+            ]),
+    );
+
     // Store modification details for response handler.
-    response.locals.numModified =
-        existingCasesByCaseRefCombo.size - unchangedCaseIdSet.size;
+    response.locals.numModified = existingCasesByCaseRefCombo.size;
```

Comment:
Now that we're removing cases that only differed by upload -- can we use MongoDB's numModified (in the CaseController)? IIRC, `uploadId` was the only piece of information that prevented us from doing so. Rather than pass this via middleware.

I can take care of this in a separate PR if you'd like.

Reply:
Done.